qutip/solver/result.py

```diff
@@ -296,6 +297,42 @@ def _pre_copy(self, state):
         """
         return state.copy()
 
+    def plot_expect(self, labels=None, title=None, show=True):
+        """
+        Plot expectation values stored in the result object.
+
+        Parameters:
+        -----------
+        labels : list of str, optional
+            Labels for the expectation values.
+            If None, keys from e_ops dictionary are used.
+        title : str, optional
+            Title of the plot. If None, a default title is used.
+        show : bool, optional
+            If True, the plot is displayed immediately.
+        """
+        if not hasattr(self, 'expect'):
+            raise AttributeError(
+                "No expectation values found in the result object.")
+
+        if labels is None:
+            if hasattr(self, 'e_ops') and isinstance(self.e_ops, dict):
+                labels = list(self.e_ops.keys())
+            else:
+                labels = [f'Expectation {i}' for i in range(len(self.expect))]
+
+        plt.figure()
+        for i, (label, values) in enumerate(zip(labels, self.expect)):
```

Comment:
`self.e_data` is a dict with the key of `e_ops` as keys (or 0, 1, ...) if e_ops was originally a list. And the expectations values as values. You could use it directly.